### PR TITLE
[fix][ci] Check full build artifacts licenses

### LINKED ISSUE #16783: check-binary-license.sh doesn't check Pulsar SQL licenses in CI
https://github.com/apache/pulsar/blob/52d8fe03160648f14dc196edaa87921064a1b2b2/.github/workflows/pulsar-ci.yaml#L106-L113

When build and check license in CI, we don't build with Pulsar SQL module. Thus, we can miss dependencies bump to Pulsar SQL, as in https://github.com/apache/pulsar/pull/16508#issuecomment-1194035662 and the first attempt of https://github.com/apache/pulsar/pull/16683.

I don't know whether it's required to build with `-Pcore-modules,-main`, but if we want to check binary licenses correctly, current workflow doesn't achieve the goal.

### PR DESCRIPTION
**Pull request by tisonkun**
This closes #16783.
This closes #15253.

### Documentation

- [ ] `doc` 
- [ ] `doc-required` 
- [x] `doc-not-needed` 
- [ ] `doc-complete` 

### Matching PR in forked repository

PR in forked repository: https://github.com/tisonkun/pulsar/pull/8

### CODE REVIEW

**Review by nodece — APPROVED** [review]

File: .github/workflows/pulsar-ci.yaml [hunk 1]
```diff
@@ -544,6 +544,10 @@ jobs:
           -Pmain,docker -Dmaven.test.skip=true -Ddocker.squash=true \
           -Dspotbugs.skip=true -Dlicense.skip=true -Dcheckstyle.skip=true -Drat.skip=true
 
+      # check full build artifacts licenses
+      - name: Check binary licenses
+        run: src/check-binary-license.sh ./distribution/server/target/apache-pulsar-*-bin.tar.gz
```
Comment (nodece) [inline .github/workflows/pulsar-ci.yaml:549]: I think you should move this command to `build-and-license-check` job.

Comment (tisonkun) [reply .github/workflows/pulsar-ci.yaml:549]: No. In that step, we don't have a full-build tarball. See also #15253.

**Review by RobertIndie — APPROVED** [review]

**Comment by tisonkun** [comment]
"Pulsar CI / Build Pulsar docker image" passed. I'm gonna rerun other tests later :(

**Pull request merged**